openapi_python_client/parser/properties/__init__.py

```diff
@@ -300,6 +302,22 @@ def get_lazy_imports(self, *, prefix: str) -> Set[str]:
             lazy_imports.update(inner_prop.get_lazy_imports(prefix=prefix))
         return lazy_imports
 
+    def get_discriminator_value(self, sub_model: ModelProperty) -> str:
+        """
+        Get discriminator's property value for sub_model
+        """
+        if self.discriminator:
+            if self.discriminator.mapping:
+                for property_value, schema_path in self.discriminator.mapping.items():
+                    ref_path = parse_reference_path(schema_path)
+                    if isinstance(ref_path, ParseError):
+                        raise TypeError()
```

Comment:
I try, when at all possible, not to `raise` anything since it usually results in a poor error message for the consumer—who usually has no idea why the OpenAPI Document is wrong (I should probably add a code style section to the contribution guide).

I think in this case, it might be a good idea to create a new property type `DiscriminatedProperty` or similar, since we should handle it in a few different ways:

1. We must enforce at parse time (return a `ParseError` if it fails) that every type within the union is a `ModelProperty` with a required property matching the discriminator
2. Instead of trying every inner property at runtime, we only attempt the build the one matching the discriminator.   So instead of a bunch of `try/except`, there will be an `if` statement or, better yet, a `dict` to look up the correct class.